frontend/src/components/widgets/FileUploader/UploadedFile.tsx

```diff
@@ -0,0 +1,148 @@
+import { MaterialIcon } from "components/shared/Icon"
+import ProgressBar from "components/shared/ProgressBar"
+import { Small, Kind as TextKind } from "components/shared/TextElements"
+import { ExtendedFile, FileStatuses, getSizeDisplay } from "lib/FileHelper"
+import { colors, Sizes, utilityClasses, spacing } from "lib/widgetTheme"
+
+import "./FileUploader.scss"
+
+export interface Props {
+  file: ExtendedFile
+  progress: number | undefined
+  onDelete: (id: string) => void
+}
+
+export interface FileStatusProps {
+  file: ExtendedFile
+  progress: number | undefined
+}
+
+const UploadedFileData = styled("div", {
+  display: "flex",
+  alignItems: "baseline",
+  flex: 1,
+  paddingLeft: spacing.lg,
+  overflow: "hidden",
+})
+
+const UploadedFileName = withStyleDeep(
+  styled("div", {
+    marginRight: spacing.sm,
+    marginBottom: spacing.xxs,
+  }),
+  utilityClasses.ellipsis
+)
+
+const StyledUploadedFile = styled("div", {
+  display: "flex",
+  alignItems: "center",
+  marginBottom: spacing.xxs,
+})
+
+export const ErrorMessage = styled("span", {
+  marginRight: spacing.xxs,
+})
+
+export const FileIcon = styled("div", {
+  display: "flex",
+  padding: spacing.xxs,
+  color: colors.secondary,
+})
+
+export const FileStatus = ({
+  file,
+  progress,
+}: FileStatusProps): React.ReactElement | null => {
+  if (progress) {
+    return (
+      <ProgressBar
+        value={progress}
+        size={Sizes.SMALL}
+        overrides={{
+          Bar: {
+            style: {
+              marginLeft: 0,
+              marginTop: "4px",
+            },
+          },
+        }}
+      />
+    )
+  }
+
+  if (file.status === FileStatuses.ERROR) {
+    return (
+      <Small className="fileError" kind={TextKind.DANGER}>
+        <ErrorMessage>{file.errorMessage || "error"}</ErrorMessage>
+        <MaterialIcon icon="error" />
+      </Small>
+    )
+  }
+
+  if (file.status === FileStatuses.UPLOADED) {
+    return (
+      <Small kind={TextKind.SECONDARY}>{getSizeDisplay(file.size, "b")}</Small>
+    )
+  }
+
+  if (file.status === FileStatuses.DELETING) {
+    return <Small kind={TextKind.SECONDARY}>Removing file</Small>
+  }
+
+  return null
+}
+
+const UploadedFile = ({
+  file,
+  progress,
+  onDelete,
+}: Props): React.ReactElement => {
+  const handleDelete = (event: React.SyntheticEvent<HTMLElement>): void => {
+    const { id } = event.currentTarget
+    onDelete(id)
+  }
+
+  return (
+    <StyledUploadedFile className="uploadedFile">
+      <FileIcon>
+        <MaterialIcon
+          type="outlined"
+          icon="insert_drive_file"
+          size={Sizes.MEDIUM}
+        />
+      </FileIcon>
+      <UploadedFileData className="uploadedFileData">
+        <UploadedFileName className="uploadedFileName" title={file.name}>
+          {file.name}
+        </UploadedFileName>
+        <FileStatus file={file} progress={progress} />
+      </UploadedFileData>
+      <Button id={file.id} onClick={handleDelete} kind={Kind.MINIMAL}>
```

Comment:
I'd probably do an inline function `({ currentTarget } ) => onDelete(currentTarget.id)` also why not `target`? Seems like the better option.

If that's the case, we can then remove the `id` prop on `Button`